webserver/templates/user/scraper.js

```diff
@@ -165,77 +175,76 @@ function reportScrobbles(struct) {
     xhr.timeout = 10 * 1000; // 10 seconds
     xhr.onload = function(content) {
         if (this.status >= 200 && this.status < 300) {
-            numCompleted++;
             console.log("successfully reported page");
+            pageDone();
         } else if (this.status >= 400 && this.status < 500) {
             times4Error++;
             // We mark 4xx errors as completed because we don't
             // retry them
-            numCompleted++;
             console.log("4xx error, skipping");
+            pageDone();
         } else if (this.status >= 500) {
             console.log("received http error " + this.status + " req'ing");
             times5Error++;
-            enqueueReport(struct);
+            reportScrobbles(struct);
+        } else {
+            console.log("received http status " + this.status + ", skipping");
+            pageDone();
         }
         if (numCompleted >= numberOfPages) {
             updateMessage("<i class='fa fa-check'></i> Import finished<br><span style='font-size:8pt'>Thank you for using ListenBrainz</span>");
         } else {
             updateMessage("<i class='fa fa-cog fa-spin'></i> Sending page " + numCompleted + " of " + numberOfPages + " to ListenBrainz<br><span style='font-size:8pt'>Please don't navigate while this is running</span>");
         }
-        getNextPageIfSlots();
     };
     xhr.ontimeout = function(context) {
         console.log("timeout, req'ing");
-        enqueueReport(struct);
+        reportScrobbles(struct);
     }
     xhr.onabort = function(context) {
         console.log("abort, req'ing");
-        enqueueReport(struct);
+        reportScrobbles(struct);
     };
     xhr.onerror = function(context) {
         console.log("error, req'ing");
-        enqueueReport(struct);
+        reportScrobbles(struct);
     };
-    xhr.onloadend = function(context) {
-        activeSubmissions--;
```

Comment:
This was problematic, because `loadend` doesn't fire until after the other handlers (so `activeSubmissions` would get checked again before it got decremented here). Anyway, `pageDone` is now the only place where this gets decremented.